login/service.go

```diff
@@ -136,8 +139,7 @@ func (keycloak *KeycloakOAuthProvider) Perform(ctx *app.LoginLoginContext, confi
 			"state":          state,
 			"known_referrer": knownReferrer,
 		}, "exchanged code to access token")
-
-		_, usr, err := keycloak.CreateOrUpdateKeycloakUser(keycloakToken.AccessToken, ctx, profileEndpoint)
+		_, usr, err := keycloak.CreateOrUpdateKeycloakUser(keycloakToken.AccessToken, ctx, profileEndpoint, WITEndpointUserProfile)
 		if err != nil {
```

Comment:
Not this err can be UnauthorizedError which came from WIT. But below we treat all UnauthorizedErrors as unapproved users. We have to handle it better and not redirect to NotApprovedURL if it came from WIT.

Reply:
I've taken out the wit interactions outside this since it didn't feel right to have that in there.